### PR TITLE
Add EmailNotification event (GSI 42)

### PR DESCRIPTION
**Pull request by TheByronHimes**
Also includes updates to the dependencies and template files

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -46,7 +46,7 @@ python_requires = >= 3.9
 
 [options.extras_require]
 dev =
-    ghga-service-chassis-lib[dev]==0.17.0
+    ghga-service-chassis-lib[dev]==0.17.7
 all =
     %(dev)s
```
Comment (mephenor) [inline setup.cfg:51]:
```suggestion
```

You can probably remove that as most of the stuff should be in `requirements-dev-common.txt` now.

File: ghga_event_schemas/pydantic_.py [hunk 2]
```diff
@@ -260,6 +260,30 @@ class Config:
         title = "file_download_served"
 
 
+class EmailNotification(BaseModel):
+    """This event is emitted by services that desire to send an email notification.
+    It is picked up by the notification service."""
+
+    to_: EmailStr = Field(..., description="The primary recipient of the email")
```
Comment (mephenor) [inline ghga_event_schemas/pydantic_.py:267]: Why the underscore, pylint thinking this is to short?
I'd rather keep is at `to` and mute the warning.

According to https://docs.pydantic.dev/usage/types/ `EmailStr` needs https://github.com/JoshData/python-email-validator. Is this pulled in somewhere or covered by the pydantic[email] dependency?

**Review by mephenor — CHANGES_REQUESTED** [review]
Some small changes, e.g. chassis-lib can probably be removed completely right now.

File: ghga_event_schemas/pydantic_.py (hunk 2 above)
Comment (TheByronHimes) [reply ghga_event_schemas/pydantic_.py:267]: I had a moment where I thought to was a reserved word. I changed the to_ to to.
And yes, the [email] pulls in the email-validator. See https://docs.pydantic.dev/install/

File: ghga_event_schemas/pydantic_.py [hunk 3]
```diff
@@ -301,6 +301,30 @@ class Config:
...
```
Comment (KerstenBreuer) [inline ghga_event_schemas/pydantic_.py:308]:
```suggestion
    recipient_email: EmailStr = Field(..., description="The primary recipient of the email")
```

File: ghga_event_schemas/pydantic_.py [hunk 4]
```diff
@@ -301,6 +301,30 @@ class Config:
         title = "file_download_served"
 
 
+class EmailNotification(BaseModel):
+    """This event is emitted by services that desire to send an email notification.
+    It is picked up by the notification service."""
```
Comment (KerstenBreuer) [inline ghga_event_schemas/pydantic_.py:306]:
```suggestion
class Notification(BaseModel):
    """This event is emitted by services that desire to send a notification e.g. via email.
    It is picked up by the notification service."""
```

File: ghga_event_schemas/pydantic_.py [hunk 5]
```diff
@@ -301,6 +301,30 @@ class Config:
...
+    recipient_name: str = Field(
+        ...,
+        description="The full name of the recipient to be used in the greeting section",
+    )
+    plaintext_body: str = Field(..., description="The basic text for the email body")
+
+    class Config:
+        """Additional Model Config."""
+
+        title = "email_notification"
```
Comment (KerstenBreuer) [inline ghga_event_schemas/pydantic_.py:325]:
```suggestion
        title = "notification"
```

File: ghga_event_schemas/pydantic_.py [hunk 6]
```diff
@@ -313,4 +337,5 @@ class Config:
     "non_staged_file_requested": NonStagedFileRequested,
     "file_staged_for_download": FileStagedForDownload,
     "file_download_served": FileDownloadServed,
+    "email_notification": EmailNotification,
```
Comment (KerstenBreuer) [inline ghga_event_schemas/pydantic_.py:340]:
```suggestion
    "notification": Notification,
```

File: ghga_event_schemas/pydantic_.py [hunk 7]
```diff
@@ -301,6 +301,30 @@ class Config:
...
+    """This event is emitted by services that desire to send an email notification.
+    It is picked up by the notification service."""
+
+    to_: EmailStr = Field(..., description="The primary recipient of the email")
+    cc: list[EmailStr] = Field(
+        default=[], description="The list of recipients cc'd on the email"
+    )
+    bcc: list[EmailStr] = Field(
+        default=[], description="The list of recipients bcc'd on the email"
+    )
```
Comment (KerstenBreuer) [inline ghga_event_schemas/pydantic_.py:314]:
```suggestion
    email_cc: list[EmailStr] = Field(
        default=[], description="The list of recipients cc'd on the email"
    )
   email_bcc: list[EmailStr] = Field(
        default=[], description="The list of recipients bcc'd on the email"
    )
```

File: ghga_event_schemas/pydantic_.py [hunk 8]
```diff
@@ -301,6 +301,30 @@ class Config:
...
+    subject: str = Field(..., description="The subject line for the email")
```
Comment (KerstenBreuer) [inline ghga_event_schemas/pydantic_.py:315]:
```suggestion
    subject: str = Field(..., description="The subject line for the notification")
```

File: ghga_event_schemas/pydantic_.py [hunk 9]
```diff
@@ -301,6 +301,30 @@ class Config:
...
+    recipient_name: str = Field(
+        ...,
+        description="The full name of the recipient to be used in the greeting section",
+    )
+    plaintext_body: str = Field(..., description="The basic text for the email body")
```
Comment (KerstenBreuer) [inline ghga_event_schemas/pydantic_.py:320]:
```suggestion
    plaintext_body: str = Field(..., description="The basic text for the notification body")
```

**Review by KerstenBreuer — APPROVED** [review]
Looks good, just some naming recommendation. But nothing essential.

File: ghga_event_schemas/pydantic_.py (hunk 4 above)
Comment (TheByronHimes) [reply ghga_event_schemas/pydantic_.py:306]: If we were ever to implement other channels of notification, the event structure would have to be changed. Like if we added SMS support we'd need a telephone number field. Then things like the subject would not be applicable and we'd need to check different fields in the event payload to figure out if we're supposed to send an email, a text, or both, and we'd lose that control provided by the required field. 
Please let me know what you think.

**Review by KerstenBreuer — APPROVED** [review]

**Review by mephenor — APPROVED** [review]

**Pull request merged**